### DISCUSSION TITLE
IDE configuration

### DISCUSSION THREAD

**Discussion by huwdp** [lightspark/lightspark#599] [General]
Hi everyone!

Currently I use Atom, Terminal, gcc and gdb to help with Lightspark development. For testing, I use FlashDevelop and Flex to create little tests for the things I implement. I was wonder, what IDE and setup you you folks use?

I find GDB a bit tedious and would like to use something with a GUI on-top.

**Comment by huwdp** [comment]
@dbluelle @eiglow

**Comment by eiglow** [comment]
At the moment I mostly just use kwrite for text editing, I would probably use something more advanced if I wanted to try and learn the codebase more. I also use KDE Dolphin's integrated terminal a lot when bisecting and stuff.

**Comment by dbluelle** [comment]
I primarily use QtCreator for programming (I also use it in my job, so I'm already familiar with it).
It's mainly leaning towards Qt-based projects, but it has a good integration for general C/C++ projects and integrated support for everything I need for lightspark (cmake/gdb/git/valgrind)
To inspect swf files that are broken in lightspark I use the JPEXS flash decompiler (see https://github.com/jindrapetrik/jpexs-decompiler)

**Comment by huwdp** [comment]
Cool, I use QtCreator for personal projects but haven't used it yet as it's quite Qt leaning. I'll starting using that
